src/tr1/game/savegame/savegame_bson.c

```diff
@@ -852,6 +853,38 @@ static bool M_LoadLOT(JSON_OBJECT *lot_obj, LOT_INFO *lot)
     return true;
 }
 
+static bool M_LoadFlares(JSON_ARRAY *const flares_arr)
+{
+    if (flares_arr == nullptr) {
+        LOG_ERROR("Malformed save: invalid or missing flares array");
+        return false;
+    }
```

Comment:
I think this will cause the BSON saves we have RN to not work, right?
I'd just return early in this case.